src/main/java/dev/openfeature/sdk/hooks/logging/LoggingHook.java

```diff
@@ -0,0 +1,92 @@
+package dev.openfeature.sdk.hooks.logging;
+
+import dev.openfeature.sdk.ErrorCode;
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.FlagEvaluationDetails;
+import dev.openfeature.sdk.Hook;
+import dev.openfeature.sdk.HookContext;
+import dev.openfeature.sdk.exceptions.OpenFeatureError;
+import java.util.Map;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.spi.LoggingEventBuilder;
+
+/**
+ * A hook for logging flag evaluations.
+ * Useful for debugging.
+ * Flag evaluation data is logged at debug and error in before/after stages and error stages, respectively.
+ */
+@Slf4j
+@edu.umd.cs.findbugs.annotations.SuppressFBWarnings(value = "RV_RETURN_VALUE_IGNORED",
+    justification = "we can ignore return values of chainables (builders) here")
+public class LoggingHook implements Hook<Object> {
```

Comment:
yes it is a logging hook, but should we be a little bit more specific, like this is an slf4j Logger hook - there can be other logging hooks for different kind of logging wrappers etc.

Reply:
I don't think I want to put `SLF4j` or anyhting like that in the class name, since slf4j is the framework the SDK uses, it's taken for granted. I will add something about the structured logging to the javadoc though.